client/extensions/woocommerce/components/query-locations/index.js

```diff
@@ -0,0 +1,65 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { fetchLocations } from 'woocommerce/state/sites/locations/actions';
+import { areLocationsLoaded } from 'woocommerce/state/sites/locations/selectors';
+
+class QueryLocations extends Component {
+	static propTypes = {
+		fetchLocations: PropTypes.func,
+		loaded: PropTypes.bool,
+		siteId: PropTypes.number.isRequired,
+	};
+
+	componentWillMount = () => {
+		const { siteId, loaded } = this.props;
+
+		if ( siteId && ! loaded ) {
+			this.props.fetchLocations( siteId );
+		}
+	};
+
+	componentWillReceiveProps = ( { siteId, loaded } ) => {
```

Comment:
Side effects should be performed in `componentDidMount` and `componentDidUpdate`. See this [cheat sheet](https://gist.github.com/bvaughn/923dffb2cd9504ee440791fade8db5f9)

Reply:
Thanks @gwwar - will fix (and open issues for other places we are doing this in woocommerce)

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/21297/commits/698f0228879b86ac921f12fd74df554610db7238